presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -200,10 +202,7 @@ public HiveWriterFactory(
         this.sortedWritingTempStagingPathEnabled = isTemporaryStagingDirectoryEnabled(session);
         this.sortedWritingTempStagingPath = getTemporaryStagingDirectoryPath(session);
         this.immutablePartitions = immutablePartitions;
-        this.insertExistingPartitionsBehavior = HiveSessionProperties.getInsertExistingPartitionsBehavior(session);
-        if (immutablePartitions) {
-            checkArgument(insertExistingPartitionsBehavior != InsertExistingPartitionsBehavior.APPEND, "insertExistingPartitionsBehavior cannot be APPEND");
-        }
+        this.insertExistingPartitionsBehavior = verifyAndGet(getInsertExistingPartitionsBehavior(session), immutablePartitions);
```

Comment:
`verifyAndGet` is not needed here. It is already checked in `getInsertExistingPartitionsBehavior`

Reply:
I believe that it should be checked by decoder, however I'm not sure. I thought that removing this code could be too invasive. In case it could be deleted I'd remove other checks in code `checkState(!immutablePartitions);` 
What do you think?

Reply:
Ok, verifyAndGet removed. `checkState(!immutablePartitions);` removed as well.